From 1c9001e9e21f56b7f15c043799547309304a9775 Mon Sep 17 00:00:00 2001
From: Leonard Crestez <leonard.crestez@nxp.com>
Date: Mon, 11 Jun 2018 18:41:46 +0300
Subject: [PATCH 3967/5242] MLK-18569 thermal: imx_sc: Fix interpreting tenths
 as millicelsius

commit  4a3d636a2a54296189fba576bb3bd7154a4ee942 from
https://source.codeaurora.org/external/imx/linux-imx.git

Linux expects millicelsius but tenths are handled incorrectly.

Fixes: 10a2548b8b60 ("MLK-14972-02 driver: thermal: Add i.MX8QM/QXP thermal support")

Signed-off-by: Leonard Crestez <leonard.crestez@nxp.com>
Acked-by: Anson Huang <anson.huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/thermal/imx_sc_thermal.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/thermal/imx_sc_thermal.c b/drivers/thermal/imx_sc_thermal.c
index 9b0ea8d..931fa7c 100644
--- a/drivers/thermal/imx_sc_thermal.c
+++ b/drivers/thermal/imx_sc_thermal.c
@@ -77,7 +77,7 @@ static int imx_sc_tsens_get_temp(void *data, int *temp)
 			return -EINVAL;
 		}
 	}
-	*temp = celsius * 1000 + tenths;
+	*temp = celsius * 1000 + tenths * 100;
 
 	return 0;
 }
-- 
1.7.9.5

